x/ecocredit/server/proposal.go

```diff
@@ -0,0 +1,45 @@
+package server
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+
+	"github.com/regen-network/regen-ledger/x/ecocredit/core"
+	"github.com/regen-network/regen-ledger/x/ecocredit/marketplace"
+)
+
+// ProposalKeeper defines methods for ecocredit gov handlers.
+type ProposalKeeper interface {
+	AddCreditType(ctx sdk.Context, ctp *core.CreditTypeProposal) error
+	AddAskDenom(ctx sdk.Context, proposal *marketplace.AllowedDenomProposal) error
+}
+
+func (s serverImpl) AddAskDenom(ctx sdk.Context, proposal *marketplace.AllowedDenomProposal) error {
+	return s.marketplaceKeeper.AddAllowedDenom(ctx, proposal)
+}
```

Comment:
Either `AddAllowedDenom` or `AllowDenom` depending on what you do above.

Reply:
[updated](https://github.com/regen-network/regen-ledger/pull/1072/commits/d701ec57ad94a754739581df1168d427fb24c47d#diff-28741a950b84cb11557badbc915aa23382c84990a80a9e6d9d6f474360743a07R15)